src/main/java/uk/ac/cam/cl/dtg/segue/dos/PgUserPreferenceManager.java

```diff
@@ -86,37 +88,48 @@ public UserPreference getUserPreference(String preferenceType, String preference
         Validate.notBlank(preferenceName);
 
         Map<Long, UserPreference> usersPreferenceMap = Maps.newHashMap();
-        try (Connection conn = database.getDatabaseConnection()) {
-            PreparedStatement pst;
-            StringBuilder sb = new StringBuilder();
-            sb.append("SELECT * FROM user_preferences WHERE user_id IN (");
 
-            for (int i = 0; i < users.size(); i++) {
-                sb.append("?").append(i < users.size() - 1 ? ", " : "");
-            }
-            sb.append(") AND preference_type=? AND preference_name=? ORDER BY user_id ASC;");
+        int pageSize = 10000;
+        int fromIndex = 0;
+        int toIndex = min(pageSize, users.size());
 
-            pst = conn.prepareStatement(sb.toString());
-            for (int i = 1; i <= users.size(); i++) {
-                pst.setLong(i, users.get(i - 1).getId());
-            }
-            pst.setString(users.size() + 1, preferenceType);
-            pst.setString(users.size() + 2, preferenceName);
+        while (fromIndex < toIndex) {
 
-            ResultSet results = pst.executeQuery();
+            List<RegisteredUserDTO> pagedUsers = users.subList(fromIndex, toIndex);
+            try (Connection conn = database.getDatabaseConnection()) {
+                PreparedStatement pst;
+                StringBuilder sb = new StringBuilder();
+                sb.append("SELECT * FROM user_preferences WHERE user_id IN (");
 
-            while (results.next()) {
-                Long userId = results.getLong("user_id");
-                UserPreference pref = new UserPreference(userId, results.getString("preference_type"),
-                        results.getString("preference_name"), results.getBoolean("preference_value"));
-                usersPreferenceMap.put(userId, pref);
-            }
+                for (int i = 0; i < pagedUsers.size(); i++) {
+                    sb.append("?").append(i < pagedUsers.size() - 1 ? ", " : "");
+                }
+                sb.append(") AND preference_type=? AND preference_name=? ORDER BY user_id ASC;");
 
-            return usersPreferenceMap;
+                pst = conn.prepareStatement(sb.toString());
+                for (int i = 1; i <= pagedUsers.size(); i++) {
+                    pst.setLong(i, pagedUsers.get(i - 1).getId());
+                }
+                pst.setString(pagedUsers.size() + 1, preferenceType);
+                pst.setString(pagedUsers.size() + 2, preferenceName);
 
-        } catch (SQLException e) {
-            throw new SegueDatabaseException("Postgres exception", e);
+                ResultSet results = pst.executeQuery();
+
+                while (results.next()) {
+                    Long userId = results.getLong("user_id");
+                    UserPreference pref = new UserPreference(userId, results.getString("preference_type"),
```

Comment:
Maybe the construction of these objects could be refactored out in to a common method? Just all these repeated string literals could cause some problems later if they need to change.